parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ValidatorTest.java

```diff
@@ -101,4 +110,117 @@
         + "FROM `GHI` AS `GHI`";
     sql(sql).rewritesTo(expected);
   }
+
+  @Test public void testCreateProcedureBeginEndLabel() {
+    String sql = "create procedure foo()\n"
+        + "label1: begin\n"
+        + "leave label1;\n"
+        + "end";
+    SqlValidator validator = getTester().getValidator();
```

Comment:
nit: Suggest putting lines 119 and 120 in a private helper method which just takes the `sql` as a parameter and returns a `SqlNode` (e.g. could also be named `parseAndValidate()`). Same for other tests too.